src/processor/snapshot/mod.rs

```diff
@@ -1,132 +1,106 @@
-use std::{collections::HashMap, fmt, fs, path::PathBuf, str::FromStr};
+use std::{fs, path::PathBuf, str::FromStr};
 
 mod bytecode;
+mod database;
 mod types;
 
 use async_trait::async_trait;
 use blake2::{Blake2s256, Digest};
 use ethers::types::{Address, H256, U256, U64};
 use eyre::Result;
-use indexmap::IndexSet;
 use state_reconstruct_fetcher::{
     constants::{ethereum, storage},
     types::CommitBlockInfoV1,
 };
 use tokio::sync::mpsc;
 
-use self::types::{SnapshotFactoryDependency, SnapshotStorageLog, StorageKey, StorageValue};
+use self::{
+    database::SnapshotDB,
+    types::{SnapshotFactoryDependency, SnapshotStorageLog},
+};
 use super::Processor;
-use crate::processor::snapshot::types::MiniblockNumber;
+use crate::processor::snapshot::types::{MiniblockNumber, StorageValue};
 
-// NOTE: What file extension to use?
-const DEFAULT_EXPORT_PATH: &str = "snapshot_export";
+const DEFAULT_DB_PATH: &str = "snapshot_db";
 
-pub struct SnapshotExporter {
-    storage_log_entries: HashMap<StorageKey, SnapshotStorageLog>,
-    factory_deps: Vec<SnapshotFactoryDependency>,
-    index_to_key_map: IndexSet<U256>,
-    path: PathBuf,
+pub struct SnapshotBuilder {
+    database: SnapshotDB,
 }
 
-impl SnapshotExporter {
-    pub fn new(path: Option<String>) -> Self {
-        let path = match path {
+impl SnapshotBuilder {
+    pub fn new(db_path: Option<String>) -> Self {
+        let db_path = match db_path {
             Some(p) => PathBuf::from(p),
-            None => PathBuf::from(DEFAULT_EXPORT_PATH),
+            None => PathBuf::from(DEFAULT_DB_PATH),
         };
 
-        let mut index_to_key_map = IndexSet::new();
-        let mut storage_log_entries = HashMap::new();
-
-        reconstruct_genesis_state(
-            &mut storage_log_entries,
-            &mut index_to_key_map,
-            storage::INITAL_STATE_PATH,
-        )
-        .unwrap();
-
-        Self {
-            storage_log_entries,
-            factory_deps: vec![],
-            index_to_key_map,
-            path,
-        }
-    }
-}
-
-impl fmt::Display for SnapshotExporter {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let mut s = String::new();
+        let mut database = SnapshotDB::new(db_path).unwrap();
 
-        for entry in &self.storage_log_entries {
-            s.push_str(&entry.1.to_string());
-            s.push('\n');
-        }
+        reconstruct_genesis_state(&mut database, storage::INITAL_STATE_PATH).unwrap();
 
-        write!(f, "{s}")
+        Self { database }
     }
 }
 
 #[async_trait]
-impl Processor for SnapshotExporter {
+impl Processor for SnapshotBuilder {
     async fn run(mut self, mut rx: mpsc::Receiver<CommitBlockInfoV1>) {
-        // TODO: Send from fetcher.
+        // TODO: Send these from fetcher.
+        let miniblock_number = U64::from(0);
```

Comment:
I believe this part was adjusted in a previous PR, so you might need to rebase on `main` again.